shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -87,3 +63,152 @@ async fn passthrough_standard() {
         10
     );
 }
+
+async fn test_create_index(client: &OpenSearch) {
+    client
+        .indices()
+        .create(IndicesCreateParts::Index("test-index"))
+        .send()
+        .await
+        .unwrap();
+
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&["test-index"]))
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(exists_response.status_code(), StatusCode::OK);
+}
+
+async fn test_index_and_search_document(client: &OpenSearch) -> String {
+    client
+        .index(IndexParts::Index("test-index"))
+        .body(json!({
+            "name": "John",
+            "age": 30
+        }))
+        .refresh(Refresh::WaitFor)
+        .send()
+        .await
+        .unwrap();
+
+    let response = client
+        .search(SearchParts::Index(&["test-index"]))
+        .from(0)
+        .size(10)
+        .body(json!({
+            "query": {
+                "match": {
+                    "name": "John",
+                }
+            }
+        }))
+        .send()
+        .await
+        .unwrap();
+
+    let results = response.json::<Value>().await.unwrap();
+    assert!(results["took"].as_i64().is_some());
```

Comment:
a few improvements here:
```rust
    assert!(results["took"].is_i64());
    let hits = results["hits"]["hits"].as_array().unwrap();
    assert_eq!(
        hits.iter().map(|x| &x["_source"]).collect::<Vec<_>>(),
        vec!(&json!({
            "name": "John",
            "age": 30,
        }))
    );
    hits[0]["_id"].as_str().unwrap().to_owned()
```

* We can use `is_i64()` instead of `as_i64().is_some()`
* We can assert a list of expected `_sources` rather than just the length.
* We can use `["hits]` directly on a `Value` instead of calling `.as_object()`